app/lib/cognito.server.ts

```diff
@@ -58,6 +69,55 @@ export function extractAttributeRequired(
   return value
 }
 
+export async function getCognitoUserFromSub(sub: string) {
+  const escapedSub = sub.replaceAll('"', '\\"')
+  const user = (
+    await cognito.send(
+      new ListUsersCommand({
+        UserPoolId,
+        Filter: `sub = "${escapedSub}"`,
+      })
+    )
+  )?.Users?.[0]
+
+  if (!user?.Username)
+    throw new Response('Requested user does not exist', {
+      status: 400,
+    })
```

Comment:
This is copied from endorsements.sever.ts. Please refactor. Don't duplicate code.